graphic.js

```diff
@@ -0,0 +1,129 @@
+var computerResultElem = document.getElementById('js-computerResult');
+
+newGameBtn.addEventListener('click', newGame);
+pickRock.addEventListener('click', function() { playerPick('rock') });
+pickPaper.addEventListener('click', function() { playerPick('paper') });
+pickScissors.addEventListener('click', function() { playerPick('scissors') });
+
+    var gameState = 'notStarted';  //started // ended
+    player = {
+        name: '',
+        score: 0
+    };
+    computer = {
+        score: 0
+    };
+    
+function setGameElements() {
+
+  switch(gameState) {
+    case 'started':
+        newGameElem.style.display = 'none';
+        pickElem.style.display = 'block';
+        resultsElem.style.display = 'block';
+      break;
+    case 'ended':
+        newGameBtn.innerText = 'Jeszcze raz';
+    case 'notStarted':
+    default:
+        newGameElem.style.display = 'block';
+        pickElem.style.display = 'none';
+        resultsElem.style.display = 'none';
+  }
+}
+    
+function newGame() {
+  player.name = prompt('Graczu, wpisz swoje imię', 'imię gracza');
+  if (player.name === null) {
+    return;
+  }
+  endPoints = prompt('Do ilu chcesz grać?');
+  if (endPoints === null || isNaN(endPoints)) {
+    return;
+  }
+
+
+  if (player.name) {
+    player.score = computer.score = 0;
+    gameState = 'started';
+    setGameElements();
+
+    playerNameElem.innerHTML = player.name;
+        setGamePoints();
+  }
+}
+
+function setGamePoints() {
+    playerPointsElem.innerHTML = player.score;
+    computerPointsElem.innerHTML = computer.score;
+}
+
+function playerPick(playerPick) {
+    var computerPick = getComputerPick();
+    
+    playerPickElem.innerHTML = playerPick;
+    computerPickElem.innerHTML = computerPick;
+    checkRoundWinner(playerPick, computerPick);
+}
+
+function getComputerPick() {
+    var possiblePicks = ['rock', 'paper', 'scissors'];
+    return possiblePicks[Math.floor(Math.random()*3)];  
+}
+
+function checkRoundWinner(playerPick, computerPick) {
+  playerResultElem.innerHTML = computerResultElem.innerHTML = '';
+
+  var winnerIs = 'player';
+
+    if (playerPick == computerPick) {
+        winnerIs = 'noone'; // remis
+    } else if (
+        (computerPick == 'rock' &&  playerPick == 'scissors') ||
+        (computerPick == 'scissors' &&  playerPick == 'paper') ||
+        (computerPick == 'paper' &&  playerPick == 'rock') ) {
+        
+        winnerIs = 'computer';
+    }
+
+    if (winnerIs == 'player') {
+        playerResultElem.innerHTML = "Wygrana!";
+        player.score++;
+        playerPointsElem.innerHTML = player.score;
+        if (player.score == endPoints){
+          endGamePlayer();
+        }
+    } else if (winnerIs == 'computer') {
+        computerResultElem.innerHTML = "Wygrana!";
+        computer.score++;
+        computerPointsElem.innerHTML = computer.score;
+        if (computer.score == endPoints) {
+          endGameComputer();
+        }
```

Comment:
to
```
+        if (computer.score == endPoints) {
 +          endGameComputer();
 +        }
```
 i podobny warunek kilka linijek wyżej przenieś za te ify wszystkie i za pomocą ternary operator wywołaj jedną lub drugą funkcję (zależy kto wygrał). DRY.

A najlepiej to z endGamePlayer i endGameComputer zrobić jedną funkcję, która może dostać różne parametry. Tam jest dużo DRY.